### ISSUE TITLE
Query information about clients that subscribe to the specified service - The returned value does not meet expectations

### ISSUE THREAD

**Issue by Myheartwilgn** [alibaba/nacos#12940]
Version：2.2.3~2.3.2
BugApi：Nacos Open API Doc -> /nacos/v2/ns/client/service/subscriber/list
Description：Null is returned when ip or port is null
Code：
```java
  @GetMapping("/service/subscriber/list")
  @Secured(action = ActionTypes.READ, resource = "nacos/admin")
  public Result<List<ObjectNode>> getSubscribeClientList(
          @RequestParam(value = "namespaceId", required = false, defaultValue = Constants.DEFAULT_NAMESPACE_ID) String namespaceId,
          @RequestParam(value = "groupName", required = false, defaultValue = Constants.DEFAULT_GROUP) String groupName,
          @RequestParam(value = "ephemeral", required = false, defaultValue = "true") Boolean ephemeral,
          @RequestParam("serviceName") String serviceName, @RequestParam(value = "ip", required = false) String ip,
          @RequestParam(value = "port", required = false) Integer port) {
      Service service = Service.newService(namespaceId, groupName, serviceName, ephemeral);
      Collection<String> allClientsSubscribeService = clientServiceIndexesManager
              .getAllClientsSubscribeService(service);
      ArrayList<ObjectNode> res = new ArrayList<>();
      for (String clientId : allClientsSubscribeService) {
          Client client = clientManager.getClient(clientId);
          Subscriber subscriber = client.getSubscriber(service);
          if (!Objects.equals(subscriber.getIp(), ip) || !Objects.equals(port, subscriber.getPort())) {
              continue;
          }
          ObjectNode item = JacksonUtils.createEmptyJsonNode();
          item.put("clientId", clientId);
          item.put("ip", subscriber.getIp());
          item.put("port", subscriber.getPort());
          res.add(item);
      }
      return Result.success(res);
  }
```
BugCode：
```java
    if (!Objects.equals(subscriber.getIp(), ip) || !Objects.equals(port, subscriber.getPort())) {
        continue;
    }
```

**Comment by KomachiSion** [comment]
What's your expected?

**Comment by Myheartwilgn** [comment]
> What's your expected?
when ip or port is null, don't continue. 
Now that the subscriber information is actually found, it is filtered out by "equal" when the ip or port is null, resulting in an empty subscriber list returned.

**Comment by KomachiSion** [comment]
So you want when ip or port not input, return all subscribers of this service?

**Comment by Myheartwilgn** [comment]
> So you want when ip or port not input, return all subscribers of this service?

Yes, in fact, according to the interface documentation, it should be

**Comment by KomachiSion** [comment]
Maybe add a new param to support when ip or port is null, return all infos. default same with old version.

Or in Nacos 3.0, default change to return all infos when ip or port is null.

**Comment by Myheartwilgn** [comment]
> Maybe add a new param to support when ip or port is null, return all infos. default same with old version.
> 
> Or in Nacos 3.0, default change to return all infos when ip or port is null.

Good！Thanks！

**Comment by zhouchunhai** [comment]
@i will solve it@

**Comment by zhouchunhai** [comment]
I think there is no need to add a new param.
![image](https://github.com/user-attachments/assets/66f248cc-56c0-4dd6-b352-1457efa6e07f)

**Comment by Myheartwilgn** [comment]
> I think there is no need to add a new param. ![image](https://private-user-images.githubusercontent.com/111631755/399650194-66f248cc-56c0-4dd6-b352-1457efa6e07f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4MDcyMzEsIm5iZiI6MTczNTgwNjkzMSwicGF0aCI6Ii8xMTE2MzE3NTUvMzk5NjUwMTk0LTY2ZjI0OGNjLTU2YzAtNGRkNi1iMzUyLTE0NTdlZmE2ZTA3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwMlQwODM1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDU5ODM2YmI0ZDM5Y2M4YzU4YmQ2NDE1NjUzOGMxZmE4ZmJlOGU4M2UzMGJjZjE4YTVmOGI3NzBlMWIxYTVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XPb9kZjrfH2G86wJ0S0nR148S-qvVROC_EWM6Y9pln4)

That's what I thought. According to the api, it could be empty

**Comment by mivljack** [comment]
Thank you for reaching out. It seems there’s an issue with the query for client subscription information on [Cava Menu Dallas](https://cavamenudallas.com/).)) I recommend reviewing the query parameters and ensuring the service logic and data sources are aligned. If the issue persists, please let me know, and we can look into it further.

**Comment by mivljack** [comment]
Thank you for reaching out. It seems there’s an issue with the query for client subscription information on Cava Menu Dallas. I recommend reviewing the query parameters and ensuring the service logic and data sources are aligned. If the issue persists, please let me know, and we can look into it further.https://cavamenudallas.com/

**Comment by Dario-s-j** [comment]
Ok

Em sex., 3 de jan. de 2025, 02:39, mivljack ***@***.***>
escreveu:
